packages/javascript/bh-shared-ui/src/hooks/useExploreGraph/useNodeSearch/useNodeSearch.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import { SearchValue } from '../../../store';
+import { useExploreParams } from '../../useExploreParams';
+import { nodeSearchGraphQuery } from '../queries';
+
+/* Reusable logic for syncing up a single node search field with browser query params on the Explore page. The value of the search field is tracked
+internally, and is only pushed to query params once the event handler is called by the consumer component. Direct changes to the associated query
+params will be synced back to the search field. */
+export const useNodeSearch = () => {
+    const [searchTerm, setSearchTerm] = useState<string>('');
+    const [selectedItem, setSelectedItem] = useState<SearchValue | undefined>(undefined);
+
+    const { primarySearch, searchType, setExploreParams } = useExploreParams();
+
+    const graphQuery = nodeSearchGraphQuery({ primarySearch, searchType });
+    const { data: graphData } = useQuery(graphQuery);
```

Comment:
good call keeping this! I realized that if we did use the `useExploreGraph` directly here then the `matchedNode` below may not return the node that was searched for in the case where the current searched for node doesnt return in the graph results.